cmd/agent/app/integrations.go

```diff
@@ -0,0 +1,281 @@
+	Hidden: true,
+}
+
+var freezeCmd = &cobra.Command{
+	Use:   "freeze",
+	Short: "Freeze list of installed python packages",
+	Long:  ``,
+	RunE:  freeze,
+}
+
+func getTufConfigPath() string {
+	here, _ := executable.Folder()
+	return filepath.Join(here, relTufConfigFilePath)
+}
+
+func getCommandPython() (string, error) {
+	if useSysPython {
+		return pythonBin, nil
+	}
+
+	here, _ := executable.Folder()
+	pyPath := filepath.Join(here, relPyPath)
+
+	if _, err := os.Stat(pyPath); err != nil {
+		if os.IsNotExist(err) {
+			return pyPath, errors.New("unable to find pip executable")
+		}
+	}
+
+	return pyPath, nil
+}
+
+func getConstraintsFilePath() (string, error) {
+	here, _ := executable.Folder()
+	cPath := filepath.Join(here, relConstraintsPath)
+
+	if _, err := os.Stat(cPath); err != nil {
+		if os.IsNotExist(err) {
+			return cPath, err
+		}
+	}
+
+	return cPath, nil
+}
+
+func getTUFConfigFilePath() (string, error) {
+	if _, err := os.Stat(tufConfig); err != nil {
+		if os.IsNotExist(err) {
+			return tufConfig, err
+		}
+	}
+
+	return tufConfig, nil
+}
+
+func tuf(args []string) error {
+	if !allowRoot && !authorizedUser() {
+		return errors.New("Please use this tool as the agent-running user")
+	}
+
+	if flagNoColor {
+		color.NoColor = true
+	}
+
+	err := common.SetupConfig(confFilePath)
+	if err != nil {
+		fmt.Printf("Cannot setup config, exiting: %v\n", err)
+		return err
+	}
+
+	pipPath, err := getCommandPython()
+	if err != nil {
+		return err
+	}
+	tufPath, err := getTUFConfigFilePath()
+	if err != nil && !withoutTuf {
+		return err
+	}
+
+	cmd := args[0]
+	implicitFlags := args[1:]
+	implicitFlags = append(implicitFlags, "--disable-pip-version-check")
+	args = append([]string{"-mpip"}, cmd)
+
+	// Add pip power-user flags
+	// cmd-flags go before the actual command
+	cmdFlags, err := tufCmd.Flags().GetStringSlice("cmd-flags")
+	if err == nil {
+		args = append(args, cmdFlags...)
+	}
+	args = append(args, implicitFlags...)
+
+	// Proxy support
+	proxies := config.GetProxies()
+	if proxies != nil {
+		proxyFlags := fmt.Sprintf("--proxy=%s", proxies.HTTPS)
+		args = append(args, proxyFlags)
+	}
+
+	// idx-flags go after the command and implicit flags
+	idxFlags, err := tufCmd.Flags().GetStringSlice("idx-flags")
+	if err == nil {
+		args = append(args, idxFlags...)
+	}
+
+	tufCmd := exec.Command(pipPath, args...)
+
+	if !withoutTuf {
+		tufCmd.Env = append(os.Environ(),
+			fmt.Sprintf("TUF_CONFIG_FILE=%s", tufPath),
+		)
+	}
+
+	// forward the standard output to the Agent logger
+	stdout, err := tufCmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stdout)
+		for in.Scan() {
+			fmt.Println(in.Text())
+		}
+	}()
+
+	// forward the standard error to the Agent logger
+	stderr, err := tufCmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+	go func() {
+		in := bufio.NewScanner(stderr)
+		for in.Scan() {
+			fmt.Println(color.RedString(in.Text()))
+		}
+	}()
+
+	err = tufCmd.Run()
```

Comment:
One thing we should do is to check that if `err == nil`, then that TUF did execute correctly. I'll think of good test conditions here.

Reply:
`err` is returned and if not `nil` cobra barks the error out.